pyupgrade.py

```diff
@@ -1827,14 +1931,14 @@ def fix_file(filename, args):
 
     if contents_text != contents_text_orig:
         print('Rewriting {}'.format(filename))
-        with io.open(filename, 'w', encoding='UTF-8', newline='') as f:
-            f.write(contents_text)
+        with io.open(filename, 'w', encoding='UTF-8', newline='') as f:  # type: ignore  # noqa
```

Comment:
OK this was super nasty. Had to pop into the mypy debugger to figure out what was going on. Since we use an `open()` call above with `rb`, that `f` is resolved to a BinaryIO, and here the same named variable `f` also gets resolved to `BinaryIO`. Will file a ticket upstream.

Reply:
The fix would be to use `with open(...) as fb` above or use `io.open` everywhere

Reply:
ah yeah can just use a different variable name, this is the usual "redefinition with different type" issue

Reply:
I think there is a new/specific interaction between open and io.open especially given that ond fix is to not mix them.  f is a new variable each time with its own scope right?

Reply:
nope, variables are function scoped in python (except for `except`'s `as` in python3+ -- and even that is only simulated through `del` shenanigans)

Reply:
:(